service/frontend/workflow_handler_test.go

```diff
@@ -1496,6 +1515,176 @@ func (s *workflowHandlerSuite) TestGetHistory() {
+	req1 := &historyservice.ReadHistoryBranchReverseRequest{
+		NamespaceId:            namespaceID.String(),
+		ShardId:                shardID,
+		BranchToken:            branchToken,
+		MaxEventId:             nextEventID,
+		LastFirstTransactionId: lastFirstTxnID,
+		PageSize:               3,
+		NextPageToken:          []byte{},
+	}
+	s.mockHistoryClient.EXPECT().ReadHistoryBranchReverse(gomock.Any(), req1).Return(&historyservice.ReadHistoryBranchReverseResponse{
+		HistoryEvents: []*historypb.HistoryEvent{
+			{
+				EventId:   int64(100),
+				EventType: enumspb.EVENT_TYPE_WORKFLOW_TASK_COMPLETED,
+			},
+			{
+				EventId:   int64(101),
+				EventType: enumspb.EVENT_TYPE_WORKFLOW_TASK_SCHEDULED,
+			},
+		},
+		NextPageToken: []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		Size_:         2,
+	}, nil)
+	req2 := &historyservice.ReadHistoryBranchReverseRequest{
+		NamespaceId:            namespaceID.String(),
+		ShardId:                shardID,
+		BranchToken:            branchToken,
+		MaxEventId:             nextEventID,
+		LastFirstTransactionId: lastFirstTxnID,
+		PageSize:               3,
+		NextPageToken:          []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+	}
+	s.mockHistoryClient.EXPECT().ReadHistoryBranchReverse(gomock.Any(), req2).Return(&historyservice.ReadHistoryBranchReverseResponse{
+		HistoryEvents: []*historypb.HistoryEvent{
+			{
+				EventId:   int64(101),
+				EventType: enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED,
+				Attributes: &historypb.HistoryEvent_WorkflowExecutionStartedEventAttributes{
+					WorkflowExecutionStartedEventAttributes: &historypb.WorkflowExecutionStartedEventAttributes{
+						SearchAttributes: &commonpb.SearchAttributes{
+							IndexedFields: map[string]*commonpb.Payload{
+								"CustomKeywordField":    payload.EncodeString("random-keyword"),
+								"TemporalChangeVersion": payload.EncodeString("random-data"),
+							},
+						},
+					},
+				},
+			},
+		},
+		NextPageToken: []byte{},
+		Size_:         1,
+	}, nil)
+
+	s.mockSearchAttributesProvider.EXPECT().GetSearchAttributes(gomock.Any(), false).Return(searchattribute.TestNameTypeMap, nil)
+	s.mockSearchAttributesMapperProvider.EXPECT().GetMapper(namespaceName).
+		Return(&searchattribute.TestMapper{}, nil).AnyTimes()
+
+	wh := s.getWorkflowHandler(s.newConfig())
+
+	history, token, eventId, err := wh.getHistoryReverse(
+		context.Background(),
+		metrics.NoopMetricsHandler,
+		namespaceID,
+		namespaceName,
+		we,
+		nextEventID,
+		lastFirstTxnID,
+		3,
+		[]byte{},
+		branchToken,
+	)
+	s.NoError(err)
+	s.NotNil(history)
+	s.Equal([]byte{}, token)
+	s.Equal(int64(100), eventId)
+
+	s.EqualValues("Keyword", history.Events[2].GetWorkflowExecutionStartedEventAttributes().GetSearchAttributes().GetIndexedFields()["AliasForCustomKeywordField"].GetMetadata()["type"])
+	s.EqualValues(`"random-data"`, history.Events[2].GetWorkflowExecutionStartedEventAttributes().GetSearchAttributes().GetIndexedFields()["TemporalChangeVersion"].GetData())
```

Comment:
Consider using s.Assert() for these because they are not requirements but assertions. Requirements are for things like NotNil where you can only run the subsequent assertion(s) if the condition holds.

Reply:
I followed suit from the existing `TestGetHistory()` before this test function. Should be done in separate PR